mail_tracking/models/mail_message.py

```diff
@@ -102,12 +142,58 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         mail_message_ids = {m.get('id') for m in messages if m.get('id')}
         mail_messages = self.browse(mail_message_ids)
         partner_trackings = mail_messages.tracking_status()
-        email_cc = mail_messages._get_email_cc()
+        failed_message = mail_messages._get_failed_message()
         for message_dict in messages:
             mail_message_id = message_dict.get('id', False)
             if mail_message_id:
                 message_dict.update({
                     'partner_trackings': partner_trackings[mail_message_id],
-                    'email_cc': email_cc[mail_message_id],
+                    'failed_message': failed_message[mail_message_id],
                 })
+                message_dict['partner_trackings'] = \
+                    partner_trackings[mail_message_id]
         return res
+
+    @api.model
+    def _prepare_dict_failed_message(self, message):
+        failed_trackings = message.mail_tracking_ids.filtered(
+            lambda x: x.state in self.get_failed_states())
+        failed_partners = failed_trackings.mapped('partner_id')
+        failed_recipients = failed_partners.name_get()
+        return {
+            'id': message.id,
+            'date': message.date,
+            'author_id': message.author_id.name_get()[0],
+            'body': message.body,
+            'failed_recipients': failed_recipients,
+        }
+
+    @api.multi
+    def get_failed_messages(self):
+        return [self._prepare_dict_failed_message(msg) for msg in self]
+
+    @api.multi
+    def toggle_tracking_status(self):
+        """Toggle message tracking action needed to ignore them in the tracking
+           issues filter"""
```

Comment:
Again try to make an effort to let summaries be just 1 line:

```python
"""Toggle message tracking action.

This will mark them to be (or not) ignored in the tracking issues filter.
"""
```